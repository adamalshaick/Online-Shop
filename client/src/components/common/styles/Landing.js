import styled from "styled-components";

export const Badge = styled.div`
  background-color: white;
  width: 100%;
  height: 100%;
  opacity: 0.85;
  font-family: "Montserrat", sans-serif;
`;

export const Border = styled.div`
  width: 500px;
  height: 250px;
  position: absolute;
  top: 30%;
  left: 50%;
  transform: translateX(-50%);
  padding: 1rem;
  border: white solid 2px;
`;

export const Text = styled.div`
  position: relative;
  top: 50%;
  transform: translateY(-50%);
  font-size: 28px;
`;
