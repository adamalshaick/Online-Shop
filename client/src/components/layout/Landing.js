import React, { Component } from "react";
import Link from "react-router-dom/Link";
import { PropTypes } from "prop-types";
import { connect } from "react-redux";

class Landing extends Component {
  componentDidMount() {
    if (this.props.auth.isAuthenticated) {
      this.props.history.push("/dashboard");
    }
  }
  render() {
    return (
      <div>
        LandingPage
        <Link className="btn btn-lg" to="/login">
          Login
        </Link>
        <Link className="btn btn-lg" to="/register">
          Sign Up
        </Link>
      </div>
    );
  }
}

Landing.propTypes = {
  auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps)(Landing);
