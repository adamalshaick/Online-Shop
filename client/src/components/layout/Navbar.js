import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { logoutUser } from "../../actions/authActions";
import { clearCurrentProfile } from "../../actions/profileActions";
import { StyledNavbar, List, ListElement } from "../common/styles/Navbar";

class Navbar extends Component {
  onLogoutClick(e) {
    e.preventDefault();
    this.props.clearCurrentProfile();
    this.props.logoutUser();
  }

  render() {
    const { isAuthenticated } = this.props.auth;

    const authLinks = (
      <List>
        <ListElement>
          <Link style={{ color: "black" }} className="link" to="/items">
            BROWSE ITEMS
          </Link>
        </ListElement>
        <ListElement>
          <Link style={{ color: "black" }} to="/dashboard">
            DASHBOARD
          </Link>
        </ListElement>
        <ListElement>
          <Link style={{ color: "black" }} to="/cart">
            CART
            <i className="ml-2 fas fa-shopping-cart" />
          </Link>
        </ListElement>
        <ListElement>
          <Link
            style={{ color: "black" }}
            to="/"
            onClick={this.onLogoutClick.bind(this)}
          >
            LOGOUT
          </Link>
        </ListElement>
      </List>
    );

    const guestLinks = (
      <List>
        <ListElement>
          <Link style={{ color: "black" }} className="link" to="/login">
            LOGIN
          </Link>
        </ListElement>
        <ListElement>
          <Link style={{ color: "black" }} to="/register">
            SIGN UP
          </Link>
        </ListElement>
      </List>
    );

    return (
      <StyledNavbar className="text-center">
        {isAuthenticated ? authLinks : guestLinks}
      </StyledNavbar>
    );
  }
}

Navbar.propTypes = {
  logoutUser: PropTypes.func.isRequired,
  clearCurrentProfile: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { logoutUser, clearCurrentProfile }
)(Navbar);
