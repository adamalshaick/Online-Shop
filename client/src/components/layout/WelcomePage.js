import React, { Component } from "react";
import Link from "react-router-dom/Link";
import { PropTypes } from "prop-types";
import { connect } from "react-redux";
import styled from "styled-components";
import Navbar from "./Navbar";

const Button = styled.figure`
  text-align: center;
  margin: 0.5rem;
  width: 100%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
  display: flex;
  background-color: whitesmoke;
  filter: brightness(100%);
  position: relative;
  left: 50%;
  transform: translateX(-50%);
  &:hover {
    filter: brightness(50%);
  }
  transition: filter 0.5s;
  @media (min-width: 1000px) {
    & {
      width: 70%;
    }
  }
`;

const Text = styled.figcaption`
  text-align: center;
  font-size: 1.1rem;
  width: 34%;
  margin-top: 19%;
  float: right;
  transition: all 0.5s;
  color: black;
`;

const Image = styled.img`
  width: 66%;
  height: 100%;
  float: left;
`;

class WelcomePage extends Component {
  render() {
    return (
      <>
        <Navbar />
        <div className="landing-page entry container">
          <div className="text-center">
            <h3 className="mt-5 mb-4">Welcome {this.props.auth.user.name}</h3>
            <Button>
              <Link to="/dashboard">
                <Image left src="./assets/images/rsz_dashboard.jpg" />
                <Text left> Dashboard</Text>
              </Link>
            </Button>
          </div>
          <div className="row mt-5">
            <div className="col-md-2" />
            <div className="col-md-8">
              <Link
                to="/items"
                style={{ border: "lightgrey solid 1px" }}
                className="col-lg-12 btn btn-light btn-lg"
              >
                Browse items for sale
              </Link>
              <Link
                to="/sell-item"
                style={{ border: "lightgrey solid 1px" }}
                className="col-lg-12 btn btn-light mt-4 btn-lg"
              >
                Sell Your item
              </Link>
            </div>
          </div>
        </div>
      </>
    );
  }
}

WelcomePage.propTypes = {
  auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps)(WelcomePage);
