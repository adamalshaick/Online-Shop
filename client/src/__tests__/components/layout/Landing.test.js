import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Landing } from "../../../components/layout/Landing";

const wrapper = shallow(<Landing />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
