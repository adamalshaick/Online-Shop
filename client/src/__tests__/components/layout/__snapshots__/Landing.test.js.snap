// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<div
  className="entry"
>
  <div>
    <Connect(Navbar) />
    <img
      src="./assets/images/rsz_online-shop-backgorund.jpg"
      style={
        Object {
          "filter": "grayscale(100%)",
          "width": "100%",
        }
      }
    />
    <styled.div>
      <styled.div>
        <styled.div
          className="text-center"
        >
          <strong>
            ONLINE SHOP
          </strong>
          <p
            style={
              Object {
                "fontSize": "14px",
                "fontWeight": "200",
              }
            }
          >
            <strong>
              FULL-STACK ONLINE SHOP APPLICATION
            </strong>
          </p>
        </styled.div>
      </styled.div>
    </styled.div>
    <div
      className="text-center mt-5"
    >
      <Link
        className="btn btn-lg btn-outline-dark mt-5"
        replace={false}
        to="/register"
      >
        Get Started
      </Link>
    </div>
  </div>
</div>
`;
